### PR TITLE
feat/langcode_utils

### PR DESCRIPTION
**Pull request by NeonJarbas**
adds 2 new utilities `pronounce_lang` and ` extract_langcode`

### CODE REVIEW

**Review by NeonDaniel — COMMENTED** [review]
Would maintaining tuples that are parsed into dicts at language load be easier to maintain? Just thinking about maintaining 2x dicts for each language

**Comment by ChanceNCounter** [comment]
I've been out for hours, and we all arrived so very simultaneously the timestamps made me think something was wrong with my clock.

I love it, but I agree with Daniel that the collections should be assembled from common data rather than maintaining two. The functions could return from that. I'd wanna think about it, but I don't think it would need to be localized at that point, either.

**Comment by NeonDaniel** [comment]
> I love it, but I agree with Daniel that the collections should be assembled from common data rather than maintaining two. The functions could return from that. I'd wanna think about it, but I don't think it would need to be localized at that point, either.

I think the names should remain localized (`es` should return "Spanish" for `en` and "Espan~ol" for `es`)

**Comment by ChanceNCounter** [comment]
Indeed. I meant as in the localized function wrapper. There probably doesn't need to be a distinct function in the Spanish parsers if the top level function can return the entry from common data and the lang code.

**Comment by JarbasAl** [comment]
should i just move it to a .json resource file like is being done for other utils like the normalizer? that was the direction lingua franca was going, makes it easy to support new langs by sharing code and treating things as data when algorithms can be shared. our architecture allows easy subclassing to account for edge cases as is done with the normalizer

**Comment by NeonDaniel** [comment]
json or yaml resources I think make sense for easier updates in general (this really is data, not code)

File: test/unittests/test_parse_en.py [hunk 1]
```diff
@@ -1158,5 +1159,31 @@ def test_gender(self):
...
+    def test_parse_lang2_code(self):
+        def test_with_conf(text, expected_lang, min_conf=0.5):
+            lang, conf = extract_langcode(text)
+            self.assertEqual(lang, expected_lang)
+            self.assertGreaterEqual(conf, min_conf)
+
+        test_with_conf("Brazilian Portuguese", 'pt-br', 1.0)
+        test_with_conf("American English", 'en-us', 1.0)
+
+        # fuzzy match strategy should be something like PARTIAL_TOKEN_SET_RATIO
```
Comment (JarbasAl) [inline test/unittests/test_parse_en.py:1183]: TODO port fuzzy match improvements from ovos_utils
